diff --git a/node_modules/expo-application/ios/ApplicationExceptions.swift b/node_modules/expo-application/ios/ApplicationExceptions.swift
index 84800f8..a1376d0 100644
--- a/node_modules/expo-application/ios/ApplicationExceptions.swift
+++ b/node_modules/expo-application/ios/ApplicationExceptions.swift
@@ -1,18 +1,18 @@
 import ExpoModulesCore
 
-internal class UrlDocumentDirectoryException: Exception {
+internal final class UrlDocumentDirectoryException: Exception, @unchecked Sendable {
   override var reason: String {
     "Unable to get url for document directory"
   }
 }
 
-internal class InstallationTimeException: Exception {
+internal final class InstallationTimeException: Exception, @unchecked Sendable {
   override var reason: String {
     "Unable to get installation time of this application"
   }
 }
 
-internal class DateCastException: Exception {
+internal final class DateCastException: Exception, @unchecked Sendable {
   override var reason: String {
     "Invalid date format"
   }
