diff --git a/node_modules/expo-font/ios/FontUtils.swift b/node_modules/expo-font/ios/FontUtils.swift
index 9c1198c..60fb0e7 100644
--- a/node_modules/expo-font/ios/FontUtils.swift
+++ b/node_modules/expo-font/ios/FontUtils.swift
@@ -11,7 +11,7 @@ internal func queryCustomNativeFonts() -> [String] {
 
   // [1] Get font family names for each font file
   let fontFamilies: [[String]] = fontFilePaths.compactMap { fontFilePath in
-    guard let fontUrl = Bundle.main.url(forResource: fontFilePath, withExtension: nil) as? URL else {
+    guard let fontUrl = Bundle.main.url(forResource: fontFilePath, withExtension: nil) else {
       return []
     }
     guard let fontDescriptors = CTFontManagerCreateFontDescriptorsFromURL(fontUrl as CFURL) as? [CTFontDescriptor] else {
